File: backend-v3/models/acquirerType.js
'use strict'

module.exports = (sequelize, DataTypes) => {
  const acquirerType = sequelize.define('acquirerType', {
    class: DataTypes.STRING,

    name: DataTypes.STRING,
    description: DataTypes.STRING,

    thumbnail: DataTypes.STRING,
    thumbnailGray: DataTypes.STRING,
    chartColor: DataTypes.STRING
  }, {
    timestamps: true,
    freezeTableName: true,
    deletedAt: 'archivedAt',
    paranoid: true
  })

  acquirerType.associate = (models) => {
    acquirerType.hasMany(models.acquirer, { foreignKey: 'acquirerTypeId' })
  }

  acquirerType.addScope('admin', {
    paranoid: false
  })

  acquirerType.addScope('agent', {
    paranoid: false
  })

  return acquirerType
}

File: frontend-mobile-v3/sdk/src/main/java/id/mikaapp/sdk/models/AcquirerType.kt
package id.mikaapp.sdk.models

import com.google.gson.annotations.SerializedName

data class AcquirerType(
    @SerializedName("id")
    val id: String,
    @SerializedName("class")
    val classX: String,
    /**
     * Acquirer name
     */
    @SerializedName("name")
    val name: String,
    @SerializedName("description")
    val description: Any,
    @SerializedName("thumbnail")
    /**
     * Acquirer logo
     */
    val thumbnail: String,
    @SerializedName("thumbnailGray")
    /**
     * Acquirer logo in gray
     */
    val thumbnailGray: String,
    @SerializedName("chartColor")
    val chartColor: String
)